### PR TITLE
Replace Octopus.Web.BaseUrl with Octopus.Web.ServerUri

### PR DESCRIPTION
**Pull request by benjimac93**
As per conversion with Octopus Support, this replaces all usages of Octopus.Web.BaseUrl with Octopus Cloud friendly variable Octopus.Web.ServerUri. 

This is due to Octopus.Web.BaseUrl being populated with internal pod URL eg. http://octopus-i008535-7b7f88fc4c-87wks:8080/app#/Spaces-1 on the Hosted version of Octopus.

### Step template checklist

- [x] `Id` should be a **GUID** that is not `00000000-0000-0000-0000-000000000000`
  - **NOTE** If you are modifying an existing step template, please make sure that you **do not** modify the `Id` property *(updating the `Id` will break the Library sync functionality in Octopus)*. 
- [x] `Version` should be incremented, otherwise the integration with Octopus won't update the step template correctly
- [x] Parameter names should not start with `$`
- [x] **Step template parameter names (the ones declared in the JSON, not the script body) should be prefixed with a namespace so that they are less likely to clash with other user-defined variables in Octopus** (see [this issue](https://github.com/OctopusDeploy/Issues/issues/2126)). For example, use an abbreviated name of the step template or the category of the step template).
- [x] `LastModifiedBy` field must be present, and (_optionally_) updated with the correct author
- [x] If a new `Category` has been created:
   - [x] An image with the name `{categoryname}.png` must be present under the `step-templates/logos` folder
   - [x] The `switch` in the `humanize` function in [`gulpfile.babel.js`](https://github.com/OctopusDeploy/Library/blob/master/gulpfile.babel.js#L92) must have a `case` statement corresponding to it

### CODE REVIEW

**Comment by harrisonmeister** [comment]
@benjimac93 - One immediate issue I can see with this, is that for Self-hosted systems, this would be a breaking change. The Server Uri would need to be set in order for this to work properly. It's set automatically on Cloud, but not on self-hosted instances.

**Comment by benjimac93** [comment]
@harrisonmeister - I was concerned that might be the case, been many years since I installed Octopus from scratch. 

Would Server Url be empty on a fresh self-hosted instance? If so, I could use Octostache and use ServerUri if populated or BaseUrl if it's not.

If unsure, I'll do an install in the morning to check

**Comment by harrisonmeister** [comment]
> @harrisonmeister - I was concerned that might be the case, been many years since I installed Octopus from scratch.
> 
> Would Server Url be empty on a fresh self-hosted instance? If so, I could use Octostache and use ServerUri if populated or BaseUrl if it's not.
> 
> If unsure, I'll do an install in the morning to check

I've not tested _recently_, but I'm 99% sure that it would be empty by default. I think the field was primarily introduced where you'd have multiple HA nodes, and therefore the single server wouldn't match what you'd access Octopus by (usually some front-facing Load balancer or reverse proxy like NGINX)

**Comment by benjimac93** [comment]
> > @harrisonmeister - I was concerned that might be the case, been many years since I installed Octopus from scratch.
> > Would Server Url be empty on a fresh self-hosted instance? If so, I could use Octostache and use ServerUri if populated or BaseUrl if it's not.
> > If unsure, I'll do an install in the morning to check
> 
> I've not tested _recently_, but I'm 99% sure that it would be empty by default. I think the field was primarily introduced where you'd have multiple HA nodes, and therefore the single server wouldn't match what you'd access Octopus by (usually some front-facing Load balancer or reverse proxy like NGINX)

Ran some tests. It is indeed empty for new installs

**Comment by benjimac93** [comment]
Tested #{if Octopus.Web.ServerUri}#{Octopus.Web.ServerUri}#{else}#{Octopus.Web.BaseUrl}#{/if} on both a self hosted & cloud instance successfully

File: step-templates/aws-linux-install-tentacle.json [hunk 1]
```diff
@@ -102,7 +102,7 @@
       "Name": "instrallTentacle.octoServerUrl",
       "Label": "Octo Server Url",
       "HelpText": "The server url to register the tentacle with.  Defaults to the base url",
-      "DefaultValue": "#{Octopus.Web.BaseUrl}",
+      "DefaultValue": "#{#{if Octopus.Web.ServerUri}#{Octopus.Web.ServerUri}#{else}#{Octopus.Web.BaseUrl}#{/if}}",
       "DisplaySettings": {
```
Comment (harrisonmeister) [inline step-templates/aws-linux-install-tentacle.json:106]: I think the outer `#{` and `}` in these parameters (all steps like this) is superfluous, and might not evaluate as you might expect?

Comment (benjimac93) [reply step-templates/aws-linux-install-tentacle.json:106]: You're right. Find/Replace went bad on 10 of the files.

**Review by coryreid — APPROVED** [review]
All LGTM - thanks for making these changes @benjimac93 !

**Pull request merged**